script/package.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+set -o errexit
+set -o noclobber
+set -o nounset
+set -o pipefail
+
+if [ -z "$1" ]
+  then
+    echo "Usage: $0 package_root [ compress_type ]"
+    echo "compress_type must be one of the types accepted by dpkg-deb (-Z), default to 'xz'"
+    exit 1
+fi
+
+readonly package_root="$1"
+readonly compress_type="${2:-xz}"
+
+# calculate uncompressed size in kB, rounded up
+package_root_size="$(du --bytes --summarize --exclude=DEBIAN "${package_root}"/ | awk '{print $1}')"
+readonly package_root_size
+readonly installed_size="$(( (package_root_size + 1024 - 1) / 1024 ))"
+
+# # creates control file
+mkdir -p "${package_root}/DEBIAN/"
+cat << EOF | envsubst > "${package_root}/DEBIAN/control"
+Package: ${PACKAGE}
+Version: ${VERSION}
+Installed-Size: ${installed_size}
+Architecture: ${ARCH}
+Maintainer: ${MAINTAINER}
+Depends: ${DEPENDS}
+Description: ${DESCRIPTION}
+Homepage: https://github.com/charlie-foxtrot/RTLSDR-Airband.git
+EOF
+
+# post install script to symlink default config
+cat << EOF > "${package_root}/DEBIAN/postinst"
+#!/bin/sh
+
+set -o errexit
+
+if [ ! -e /usr/share/rtl_airband/config/default.conf ]; then
+  ln --symbolic /usr/share/rtl_airband/config/basic_emergency.conf /usr/share/rtl_airband/config/default.conf
+fi
+EOF
+
+# pre remove script to delete postinst symlink
+cat << EOF > "${package_root}/DEBIAN/prerm"
+#!/bin/sh
+
+set -o errexit
+
+if [ -L /usr/share/rtl_airband/config/default.conf ]; then
+  rm /usr/share/rtl_airband/config/default.conf
+fi
+EOF
+
+chmod 755 "${package_root}/DEBIAN/postinst" "${package_root}/DEBIAN/prerm"
+
+
+# creates package structure
+mkdir --parents "${package_root}/lib/systemd/system/"
+sed 's|/usr/local/bin|/usr/bin|g' init.d/rtl_airband.service > "${package_root}/lib/systemd/system/rtl_airband.service"
+mkdir --parents "${package_root}/usr/bin"
+cp build_Release_NFM/src/rtl_airband "${package_root}/usr/bin/"
```

Comment:
only packaging the NFM version, what would it take to build / package additional versions in the future?  could the build directory be an input parameter to this script?

Reply:
We are already building all the 4 _variants_ at the same time, so it would be a matter to parameterize that line.

I picked the non-debug-NFM-enabled as I thought it was the most useful from a release point of view but I don't have a strong opinion on that.

The only downside I see packaging all those is about how to put that _variant_ information in the package name in a nice way.
I though also about including all those in the same package with different executable names, but that would work as "internal" package not as a release one.
Maybe it would make sense packaging the two non-debug variants if there is a big difference in performance by excluding NFM, but I'm not aware of it.